File: resolucoes/JavaScript/lista01/ex008.js
const prompt = require('prompt-sync')();

function main() {
    const distanciaEmMetros = Number(prompt("Digite uma distância em metros:"))      

    const km = distanciaEmMetros / 1000;
    const hm = distanciaEmMetros / 100;
    const dam = distanciaEmMetros / 10;
    const dm = distanciaEmMetros * 10;
    const cm = distanciaEmMetros * 100;
    const mm = distanciaEmMetros * 1000;

    prompt(`A distância de ${distanciaEmMetros} corresponde a: ${km}KM, ${hm}HM, ${dam}DAM, ${dm}DM, ${cm}CM, ${mm}MM`)
}   

module.exports = {
    main
}

File: resolucoes/Lua/lista01/ex008.lua
print("Digite uma distancia em metros: ")
metros = tonumber(io.read()) -- Lê a distância em metros digitada pelo usuário e converte para um número

quilometros = metros / 1000 -- Calcula a distância em quilômetros
hectometros = metros / 100 -- Calcula a distância em hectômetros
decametros = metros / 10 -- Calcula a distância em decâmetros
decimetros = metros * 10 -- Calcula a distância em decímetros
centimetros = metros * 100 -- Calcula a distância em centímetros
milimetros = metros * 1000 -- Calcula a distância em milímetros

-- Mostra a conversão na tela

print("A distancia de " .. metros .. "m corresponde a:")
print(quilometros .. "Km")
print(hectometros .. "Hm")
print(decametros .. "Dam")
print(decimetros .. "dm")
print(centimetros .. "cm")
print(milimetros .. "mm")
